library/Core.cpp

```diff
@@ -2042,6 +2041,11 @@ void Core::doUpdate(color_ostream &out, bool first_update)
     // Execute per-frame handlers
     onUpdate(out);
 
+    if (df::global::ui->main.autosave_request || strict_virtual_cast<df::viewscreen_optionst>(screen))
```

Comment:
Are you sure the options screen is a good place to save persistent data? What if the user doesn't save the game?

Reply:
If user doesn't save, histfigs won't be saved either. However, with this, will it save on every tick when on that screen?

Reply:
Then the data goes into `data/saves/current` and doesn't actually get saved. Is there a better way to detect hard saves?

Come to think of it, this won't save data when a game ends, so it probably should be a different screen. Would viewscreen_savegamest work? I don't remember why I didn't use that for df-ai.

Reply:
It's possible that data isn't in a safe state when viewscreen_savegamest is shown, although it could be safe if we hook in at the first tick. #955 hooks in to the general_ref_nemesis::write_file vmethod, which could run later, though.
Does viewscreen_savegamest display when abandoning a game (or when a fort fails)?

Reply:
Was a resolution ever reached on this point?

Reply:
Yes. The current patch uses the first tick of `viewscreen_savegamest`.

https://github.com/DFHack/dfhack/pull/1402/files#diff-ec49c48896f67f205270e832ab410b1eR2043